jib-core/src/main/java/com/google/cloud/tools/jib/api/ImageReference.java

```diff
@@ -71,8 +71,12 @@
    */
   private static final String REFERENCE_REGEX =
       String.format(
-          "^(?:(%s)/)?(%s)(?:(?::(%s))|(?:@(%s)))?$",
-          REGISTRY_REGEX, REPOSITORY_REGEX, TAG_REGEX, DescriptorDigest.DIGEST_REGEX);
+          "^(?:(%s))?(?:(%s)/)?(%s)(?:(?::(%s))|(?:@(%s)))?$",
```

Comment:
Do we need this prefix extension that is only specific to Jib? That could remain external and this class may remain general. Also feels more consistent as it doesn't parse `docker://` or `tar://`. Why only `registry://`?

Reply:
It's in `ImageReference` because of the suggestion to require the registry when using the `registry://` prefix. Although, I'm not really sure if that's something we should do, considering the conclusion we came to about following the URI spec.

If we want to be consistent we can do one of the following:
- parse all of the prefixes in `ImageReference`, but that doesn't work for `tar://` since it's a path, not an image reference.
- remove the prefix before calling `ImageReference.parse`, but then we can't tell if the registry was explicit or not without doing some extra parsing outside of `ImageReference`.
- not require an explicit registry with the `registry://` prefix.